### PR TITLE
Fix computing Timestamps in Ruby

### PR DESCRIPTION
**Pull request by vincent-psarga**
## Summary

This should provide a fix for [cucumber-ruby#1438](https://github.com/cucumber/cucumber-ruby/issues/1438)

Basically, the timestamps generated by `cucumber-messages` are wrong (the way we used the `Time.to_r` does not really correspond to a seconds/nanos division).

This solution is not perfect as there are slight discrepencies  during the conversion, less that 1e6 seconds that said, once again, slight).

I'll try to fix that too before merging.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [x] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG accordingly.

### CODE REVIEW

**Comment by vincent-psarga** [comment]
I'd like this to be merged so we can make a release and show correct summaries with `cucumber-ruby` - do you think the 10e-6 precision its enough ? (I personally think so, I'd even think milliseconds would be largely enough as a precision for this kind of tests)

File: messages/ruby/lib/cucumber/messages/time_conversion.rb [hunk 1]
```diff
@@ -23,6 +25,10 @@ def duration_to_seconds(duration)
         nanos_part = duration.nanos.to_f / NANOSECONDS_PER_SECOND
         seconds_part + nanos_part
       end
+
+      def time_to_nanos(time)
+        time.to_i * NANOSECONDS_PER_SECOND + time.nsec
```
Comment (mpkorstanje) [inline messages/ruby/lib/cucumber/messages/time_conversion.rb:30]: This might overflow. And I can't work out who might use this.

**Comment by mpkorstanje** [comment]
Aside from using ISO 8601 rather then seconds since the epoch I don't see a better way to do this. I also think that sub millisecond precision is pointless.

File: messages/ruby/lib/cucumber/messages/time_conversion.rb (hunk 1 above)
Comment (vincent-psarga) [reply messages/ruby/lib/cucumber/messages/time_conversion.rb:30]: indeed, I'll drop that :)

**Comment by vincent-psarga** [comment]
> I also think that sub millisecond precision is pointless.

I agree and don't think anyone will have use of nanoseconds anyway. I think we've inherited that from the default `Timestamp` and `Duration` provided by `protobuf` (before we added our own messages for that), but I'm not sure this will prove useful anytime soon.

**Pull request merged**